### DISCUSSION TITLE
Error installing 3.2.1 (CommandError: No installed app with label 'netbox_topology_views')

### DISCUSSION THREAD

**Discussion by chrisintheshell** [mattieserver/netbox-topology-views#251] [Q&A] [answered]
Good morning,

Running into an issue trying to install netbox_toplogy_views under Netbox version 3.3.10. I'm able to run the `pip3 install netbox-topology-views` under the venv successfully but when I run `python3 manage.py migrate netbox_topology_views` I get an error:

```
[root@vica1-netbox01 ~]# source /opt/netbox/venv/bin/activate
(venv) [root@vica1-netbox01 ~]# cd /opt/netbox/netbox
(venv) [root@vica1-netbox01 netbox]# pip3 install netbox-topology-views
Collecting netbox-topology-views
  Downloading netbox_topology_views-3.2.1-py3-none-any.whl (360 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 360.2/360.2 kB 5.9 MB/s eta 0:00:00
Installing collected packages: netbox-topology-views
Successfully installed netbox-topology-views-3.2.1
(venv) [root@vica1-netbox01 netbox]# python3 manage.py migrate netbox_topology_views
CommandError: No installed app with label 'netbox_topology_views'.
```

**Comment by mattieserver** [comment]
Hi,

Can you check if you added the following to the netbox configuration.py file:
PLUGINS = ["netbox_topology_views"]
(Or just the plugin name if you already have other plugins installed)

**Comment by chrisintheshell** [comment]
@mattieserver Thanks for the prompt response. I currently have it commented out until I finish the install. Or at least thats what I was reading in the install instructions.

Should I have set this first?

**Comment by chrisintheshell** [comment]
@mattieserver if I un-comment `PLUGINS = ["netbox_topology_views"]` in the config and then re-run `python3 manage.py migrate netbox_topology_views`, I get the following:

```
(venv) [root@vica1-netbox01 netbox]# python3 manage.py migrate netbox_topology_views
Traceback (most recent call last):
  File "/opt/netbox-3.3.10/netbox/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/netbox-3.3.10/venv/lib64/python3.9/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/opt/netbox-3.3.10/venv/lib64/python3.9/site-packages/django/core/management/__init__.py", line 420, in execute
    django.setup()
  File "/opt/netbox-3.3.10/venv/lib64/python3.9/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/opt/netbox-3.3.10/venv/lib64/python3.9/site-packages/django/apps/registry.py", line 124, in populate
    app_config.ready()
  File "/opt/netbox-3.3.10/netbox/extras/plugins/__init__.py", line 73, in ready
    menu_items = import_object(f"{self.__module__}.{self.menu_items}")
  File "/opt/netbox-3.3.10/netbox/extras/plugins/utils.py", line 31, in import_object
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/netbox-3.3.10/venv/lib64/python3.9/site-packages/netbox_topology_views/navigation.py", line 1, in <module>
    from extras.plugins import PluginMenu, PluginMenuItem
ImportError: cannot import name 'PluginMenu' from 'extras.plugins' (/opt/netbox-3.3.10/netbox/extras/plugins/__init__.py)
```

**Comment by dreng** [comment, marked as answer]
Hi,

I guess, the PluginMenu class has been introduced with NetBox v3.4. Please upgrade your NetBox installation or use an older version of Netbox Topology Views.

**Reply by mattieserver** [reply]
V3.1.1 should be the latest version of the plugin you can use if you are not on netbox 3.4. I will update the docs to reflect this.

**Reply by chrisintheshell** [reply]
Got it. That worked — thank you for your help! One small comment on the instructions is that it's not clear that `PLUGINS = ["netbox_topology_views"]` should be set first before entering the virtual environment and installing the netbox-topology-views python package.

The installation instructions make it sound like this should be set after installing netbox-topology-views.

Again, thanks for all your help and for making an awesome plugin to Netbox.
